node/src/types/chainspec/core_config.rs

```diff
@@ -14,10 +14,25 @@ use serde::{
 use casper_types::testing::TestRng;
 use casper_types::{
     bytesrepr::{self, FromBytes, ToBytes},
+    system::auction::VESTING_SCHEDULE_LENGTH_MILLIS,
     TimeDiff,
 };
 use tracing::{error, warn};
 
+fn vesting_schedule_period<'de, D: Deserializer<'de>>(
```

Comment:
Let's stick with validating in the `is_valid` method please.  Something like:
```rust
        if self.vesting_schedule_period.millis() > VESTING_SCHEDULE_LENGTH_MILLIS {
            error!(
                vesting_schedule_millis = self.vesting_schedule_period.millis(),
                max_millis = VESTING_SCHEDULE_LENGTH_MILLIS
                "vesting schedule period too long"
            );
            return false;
        }
```

Reply:
Addressed